### PR TITLE
Closes #47 - Switch to `admiraldev::assert_unit()`

### LINKED ISSUE #47: General Issue: Update `derive_param_waisthip` & `derive_param_waistheight` to use unit assertion from Admiraldev
### Background Information

The packages `derive_param_waisthip` & `derive_param_waistheight` assets units using a non-exported function `assert_unit()` from `R/assertions.R`.

This functionality should preferably be in Admiraldev.

@yurovska have created a feature request for this in admiraldev (https://github.com/pharmaverse/admiraldev/issues/468).

Also see discussions here: https://github.com/pharmaverse/admiral/issues/2513

When implemented in admiraldev, we should change our functions to use tis instead.

### Definition of Done

1. Assert units using `admiraldev::assert_unit()`
2. Remove assertions function (and file if empty) from `assertions.R`

### PR DESCRIPTION
**Pull request by yurovska**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by yurovska** [comment]
@AndersAskeland @manciniedoardo

The updated `admiraldev::assert_unit()` has now been merged to `main`, so this is to finalize the ratio functions by switching from our internal unit assertion to admiraldev's one. I'm also taking a chance to update the code as per discussion [here](https://github.com/pharmaverse/admiralmetabolic/pull/33#issuecomment-2451644641). Thanks in advance for your review.

File: R/derive_advs_params.R [hunk 1]
```diff
@@ -514,7 +524,7 @@ derive_param_ratio <- function(dataset,
 
   ### If `get_unit_expr` provided then check units and enable units conversion ----
 
-  if (!missing(get_unit_expr) && !is.null(get_unit_expr)) {
+  if (unit_conversion == TRUE && !missing(get_unit_expr) && !is.null(get_unit_expr)) {
```
Comment (manciniedoardo) [inline R/derive_advs_params.R:527]: Minor, but stylistically I'd prefer `if(unit conversion ...` rather than `if(unit_conversion == TRUE ...` throughout:

```suggestion
  if (unit_conversion && !missing(get_unit_expr) && !is.null(get_unit_expr)) {
```

**Review by manciniedoardo — APPROVED** [review]
LGTM - just one minor comment from me. I tested the function and everything works as expected. Test coverage still good too. Thanks @yurovska!

@AndersAskeland feel free to review and merge when ready.

**Review by AndersAskeland — CHANGES_REQUESTED** [review]
It is looking good!

I have one thing that I think we should discuss before merging, which is how we should handle constant variables.

For instance, as the code is now, `derive_param_waisthgt()` will set height as constant if we provide an argument for `constant_by_vars`. I fear that this might be a bit confusing, both because the documentation says "By variables for when Height is constant", and that we don't specifically state we want a constant. I actually think we should have an additional argument, where we more specifically set height as a constant. For instance, `constant_height`. What are your thoughts on this @yurovska ?

**Comment by yurovska** [comment]
> It is looking good!
> 
> I have one thing that I think we should discuss before merging, which is how we should handle constant variables.
> 
> For instance, as the code is now, `derive_param_waisthgt()` will set height as constant if we provide an argument for `constant_by_vars`. I fear that this might be a bit confusing, both because the documentation says "By variables for when Height is constant", and that we don't specifically state we want a constant. I actually think we should have an additional argument, where we more specifically set height as a constant. For instance, `constant_height`. What are your thoughts on this @yurovska ?

@AndersAskeland Thanks for your comment. I personally don't find it confusing given that all other similar `{admiral}` functions are designed in the same way. For instance, see [derive_param_bmi](https://pharmaverse.github.io/admiral/reference/derive_param_bmi.html)

**Comment by AndersAskeland** [comment]
> > It is looking good!
> > I have one thing that I think we should discuss before merging, which is how we should handle constant variables.
> > For instance, as the code is now, `derive_param_waisthgt()` will set height as constant if we provide an argument for `constant_by_vars`. I fear that this might be a bit confusing, both because the documentation says "By variables for when Height is constant", and that we don't specifically state we want a constant. I actually think we should have an additional argument, where we more specifically set height as a constant. For instance, `constant_height`. What are your thoughts on this @yurovska ?
> 
> @AndersAskeland Thanks for your comment. I personally don't find it confusing given that all other similar `{admiral}` functions are designed in the same way. For instance, see [derive_param_bmi](https://pharmaverse.github.io/admiral/reference/derive_param_bmi.html)

In that case, I agree we should keep it consistent with admiral! Great work @yurovska !
